<a href="https://colab.research.google.com/github/AdrianRamos956/Homeworks/blob/main/YZQ049_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Lab 3

In [ ]:
#This begining of the code is imporing everyhting we will need for this lab
import pandas as pd
import numpy as np
from sklearn.svm import SVC
from sklearn import metrics
from sklearn.model_selection import cross_validate
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import confusion_matrix
from sklearn import svm
#This gets the data.csv and reads it and stores it all into df
df=pd.read_csv("data.csv")
#This makes the Diagnosis form M for Malignant and B for Benign Binary so we can actually use them in these Classifications
df['diagnosis']=df['diagnosis'].map({'M':1,'B':0})
df.head()

id  diagnosis  ...  fractal_dimension_worst  Unnamed: 32
0    842302          1  ...                  0.11890          NaN
1    842517          1  ...                  0.08902          NaN
2  84300903          1  ...                  0.08758          NaN
3  84348301          1  ...                  0.17300          NaN
4  84358402          1  ...                  0.07678          NaN

[5 rows x 33 columns]

In [ ]:
#Sets X to be everything important (We exclude ID because that is not used in diagnosing someone at all)
X = df[['radius_mean', 'texture_mean', 'perimeter_mean', 'area_mean', 'smoothness_mean', 'compactness_mean', 'concavity_mean', 'concave points_mean', 'symmetry_mean','fractal_dimension_mean',
                 'radius_se', 'texture_se', 'perimeter_se', 'area_se', 'smoothness_se', 'compactness_se', 'concavity_se', 'concave points_se', 'symmetry_se', 'fractal_dimension_se',
                 'radius_worst', 'texture_worst', 'perimeter_worst', 'area_worst', 'smoothness_worst', 'compactness_worst', 'concavity_worst', 'concave points_worst', 'symmetry_worst','fractal_dimension_worst']]

#OF course out Y will be the diagnosis isself it is the yes or the no
y = df['diagnosis']

In [ ]:
I chose to use Linear Regression, SVM, and KNN because i though that these would be the best to use since they are all so versitile in what they do and how you can use them for almost any situation

Linear Regression

In [ ]:
#This is the spliter to train it nice and well
X_trainLR, X_testLR, y_trainLR, y_testLR = train_test_split(X,y,test_size=0.3,random_state=109)

#We want to scale the data so it is nice
scaler = StandardScaler()

X_trainLR = scaler.fit_transform(X_trainLR)
X_testLR = scaler.transform(X_testLR)

#Initialize Logistic regression
LR = LogisticRegression()

#We fit it
LR.fit(X_trainLR, y_trainLR)

#We predict it
y_predLR = LR.predict(X_testLR)

#This is the accuracy output
accuracy = metrics.accuracy_score(y_testLR, y_predLR)
accuracy_percentage = 100 * accuracy
accuracy_percentage

98.24561403508771

In [ ]:
#This is the Confusion Matrix the professor asked for
print(confusion_matrix(y_testLR, y_predLR))

[[108   0]
 [  3  60]]


In [ ]:
#This is the Precision, Recall, F-measures the professor asked for
print(classification_report(y_testLR, y_predLR))

              precision    recall  f1-score   support

           0       0.97      1.00      0.99       108
           1       1.00      0.95      0.98        63

    accuracy                           0.98       171
   macro avg       0.99      0.98      0.98       171
weighted avg       0.98      0.98      0.98       171



SVM

In [ ]:
#This is the spliter to train it nice and well
X_trainSV, X_testSV, y_trainSV, y_testSV = train_test_split(X,y,test_size=0.3,random_state=109)

#We want to scale the data so it is nice
scaler = StandardScaler()

X_trainSV = scaler.fit_transform(X_trainSV)
X_testSV = scaler.transform(X_testSV)

#Initialize SVM
clf = svm.SVC(kernel='linear')

#We fit it
clf.fit(X_trainSV, y_trainSV)

#We predict it
y_predSV = clf.predict(X_testSV)

#This is the accuracy output
accuracy = metrics.accuracy_score(y_testSV, y_predSV)
accuracy_percentage = 100 * accuracy
accuracy_percentage

98.83040935672514

In [ ]:
#This is the Confusion Matrix the professor asked for
print(confusion_matrix(y_testSV, y_predSV))

[[108   0]
 [  2  61]]


In [ ]:
#This is the Precision, Recall, F-measures the professor asked for
print(classification_report(y_testSV, y_predSV))

              precision    recall  f1-score   support

           0       0.98      1.00      0.99       108
           1       1.00      0.97      0.98        63

    accuracy                           0.99       171
   macro avg       0.99      0.98      0.99       171
weighted avg       0.99      0.99      0.99       171



KNN

In [ ]:
#This is the spliter to train it nice and well
X_trainKN, X_testKN, y_trainKN, y_testKN = train_test_split(X,y,test_size=0.3,random_state=109)

#We want to scale the data so it is nice
scaler = StandardScaler()

X_trainKN = scaler.fit_transform(X_trainKN)
X_testKN = scaler.transform(X_testKN)

#Initialize KNN with the default neighbors
classifier = KNeighborsClassifier(n_neighbors=5)

#We fit it
classifier.fit(X_trainKN, y_trainKN)

#We predict it
y_predKN = classifier.predict(X_testKN)

#This is the accuracy output
accuracy = metrics.accuracy_score(y_testKN, y_predKN)
accuracy_percentage = 100 * accuracy
accuracy_percentage

96.49122807017544

In [ ]:
#This is the Confusion Matrix the professor asked for
print(confusion_matrix(y_testKN, y_predKN))

[[106   2]
 [  4  59]]


In [ ]:
#This is the Precision, Recall, F-measures the professor asked for
print(classification_report(y_testKN, y_predKN))

              precision    recall  f1-score   support

           0       0.96      0.98      0.97       108
           1       0.97      0.94      0.95        63

    accuracy                           0.96       171
   macro avg       0.97      0.96      0.96       171
weighted avg       0.96      0.96      0.96       171



In the end it seems that with the given data set and with everything that i have set up everything it seems as though SVM is the best one for the Breast Cancer statistics becasue it had a Accuracy of 98.83